### PR TITLE
refactor: remove useless DenomMetadata key function

### PR DESCRIPTION
**Pull request by tkxkd0159**
Make DenomMetadata substore only with DenomMetadataPrefix

## Description

closes: #714 

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/bank/keeper/keeper.go [hunk 1]
```diff
@@ -119,7 +119,8 @@ func NewBaseKeeper(
 // WithMintCoinsRestriction restricts the bank Keeper used within a specific module to
 // have restricted permissions on minting via function passed in parameter.
 // Previous restriction functions can be nested as such:
-//  bankKeeper.WithMintCoinsRestriction(restriction1).WithMintCoinsRestriction(restriction2)
+//
+//	bankKeeper.WithMintCoinsRestriction(restriction1).WithMintCoinsRestriction(restriction2)
```
Comment (zemyblue) [inline x/bank/keeper/keeper.go:123]: the reason about this changes?

**Review by zemyblue — COMMENTED** [review]
Please add changelog.

File: x/bank/keeper/keeper.go (hunk 1 above)
Comment (tkxkd0159) [reply x/bank/keeper/keeper.go]: It was typo. I rollbacked it

**Review by loin3 — COMMENTED** [review]
Changes in https://github.com/cosmos/cosmos-sdk/pull/9890/files x/bank/migrations/v043/keys.go does not reflected here.
I'm not sure about this, is it okay not reflect this?

**Comment by tkxkd0159** [comment]
> Changes in https://github.com/cosmos/cosmos-sdk/pull/9890/files x/bank/migrations/v043/keys.go does not reflected here. I'm not sure about this, is it okay not reflect this?

yeah, it's legacy migration. So we don't need it

**Review by zemyblue — APPROVED** [review]

**Review by 0Tech — APPROVED** [review]

**Pull request merged**